pkg/networkservice/common/mechanisms/recvfd/server_test.go

```diff
@@ -0,0 +1,238 @@
+	"google.golang.org/grpc/credentials/insecure"
+
+	"github.com/networkservicemesh/sdk/pkg/networkservice/chains/client"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/common/mechanisms/recvfd"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/common/mechanisms/sendfd"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/chain"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
+	"github.com/networkservicemesh/sdk/pkg/tools/grpcfdutils"
+	"github.com/networkservicemesh/sdk/pkg/tools/grpcutils"
+	"github.com/networkservicemesh/sdk/pkg/tools/sandbox"
+)
+
+type checkRecvfdServer struct {
+	t                     *testing.T
+	onFileClosedCallbacks map[string]func()
+}
+
+func (n *checkRecvfdServer) Close(ctx context.Context, conn *networkservice.Connection) (*empty.Empty, error) {
+	return next.Server(ctx).Close(ctx, conn)
+}
+
+func (n *checkRecvfdServer) Request(ctx context.Context, request *networkservice.NetworkServiceRequest) (*networkservice.Connection, error) {
+	err := grpcfdutils.InjectOnFileReceivedCallback(ctx, n.onRecvFile)
+	require.NoError(n.t, err)
+
+	return next.Server(ctx).Request(ctx, request)
+}
+
+func (n *checkRecvfdServer) onRecvFile(fileName string, file *os.File) {
+	// checking a file is closed by recvfd by setting a finalizer
+	runtime.SetFinalizer(file, func(file *os.File) {
+		onFileClosedCallback, ok := n.onFileClosedCallbacks[fileName]
+		if ok {
+			onFileClosedCallback()
+		}
+	})
+}
+
+func createFile(fileName string, t *testing.T) (inodeURLStr string, fileClosedContext context.Context, cancelFunc func()) {
+	f, err := os.Create(fileName)
+	require.NoError(t, err, "Failed to create and open a file: %v", err)
+
+	err = f.Close()
+	require.NoError(t, err, "Failed to close file: %v", err)
+
+	fileClosedContext, cancelFunc = context.WithCancel(context.Background())
+
+	inodeURL, err := grpcfd.FilenameToURL(fileName)
+	require.NoError(t, err)
+
+	inodeURLStr = inodeURL.String()
+
+	return
+}
+
+func startServer(ctx context.Context, t *testing.T, testServerChain *networkservice.NetworkServiceServer, serveURL *url.URL) {
+	var grpcServer = grpc.NewServer(grpc.Creds(grpcfd.TransportCredentials(insecure.NewCredentials())))
+	networkservice.RegisterNetworkServiceServer(grpcServer, *testServerChain)
+
+	var errCh = grpcutils.ListenAndServe(ctx, serveURL, grpcServer)
+
+	require.Len(t, errCh, 0)
+}
+
+func createClient(ctx context.Context, u *url.URL) networkservice.NetworkServiceClient {
+	return client.NewClient(
+		ctx,
+		client.WithClientURL(sandbox.CloneURL(u)),
+		client.WithDialOptions(grpc.WithTransportCredentials(
+			grpcfd.TransportCredentials(insecure.NewCredentials())),
+		),
+		client.WithDialTimeout(time.Second),
+		client.WithoutRefresh(),
+		client.WithAdditionalFunctionality(sendfd.NewClient()))
+}
+
+func TestRecvfdClosesSingleFile(t *testing.T) {
+	t.Cleanup(func() { goleak.VerifyNone(t) })
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```

Comment:
Time limit for unit test is 1s.
```suggestion
	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
```